scopes/envs/envs/envs.docs.md

```diff
@@ -1,33 +1,797 @@
+    // ...
+
+    // The Compiler aspect
+    private compiler: CompilerMain,
+
+    // The Tester aspect
+    private tester: TesterMain
+  ) {}
+
+  getBuildPipe(): BuildTask[] {
+    return [
+      this.compiler.createTask('StencilCompiler', this.getCompiler()),
+      this.tester.task,
+    ];
+  }
+}
 ```
 
-# Configure workspace to use the env
+## Extending multiple runtime environments 
+
+An environment may operate in multiple runtime environments: 'Main', which runs on the server and 'UI' and 'Preview', which run on the browser.
+Each runtime environment runs all files that are named with its corresponding file pattern.
+
+An environment extension that runs on multiple runtimes is called "Aspect" an will have the following file structure:
 
-add this to your workspace.jsonc file.
 ```
-"teambit.envs/envs": {
-  "env": "your-new-env",
-  "config": {}
+|-- env-extension
+    |-- env-extension.main.ts
+    |-- env-extension.ui.tsx
+    |-- env-extension.preview.tsx
+    |-- env.extension.aspect.ts
+```
+
+### Registering an environment as an aspect
+
+Create a `*.aspect.ts` file:
+
+For example:
+```shell
+$ touch path/to/extension/env-extension.aspect.ts
+```
+
+Place the following lines to register your environment as a multiple runtime extension (a.k.a, an Aspect):
+
+```ts
+// env-extension.ts
+
+import { Aspect } from '@teambit/harmony';
+
+export const ReactWithProvidersAspect = Aspect.create({
+  // The ID should be your component's ID
+  // Make sure to track your extension component before registering it as an Aspect
+  id: 'my-scope.react-with-providers',
+});
+
+```
+
+### Registering a runtime extension
+An aspect is a collection of multiple extensions, each extending a specific runtime.
+
+Register each runtime extension to its corresponding runtime, using the `addRuntime` method.
+
+For example:
+
+```typescript
+// react-with-providers.preview.ts
+
+export class ReactWithProvidersPreview {
+  static runtime = PreviewRuntime;
+
+  // ...
+
+  static async provider([react]: [ReactPreview]) {
+    
+    // ...
+
+    return new ReactWithProvidersPreview();
+  }
+}
+
+ReactWithProvidersAspect.addRuntime(ReactWithProvidersPreview);
+```
+
+
+
+### Runtime environments
+
+#### Main
+
+`*.main.runtime.ts`
+
+Node files that run in a node runtime environments and outputs to the terminal.
+
+**Example:**  
+The React environment TypeScript compiler will be extended in the main runtime.
+
+```typescript
+// react-extension.main.ts
+
+const tsconfig = require('./typescript/tsconfig.json');
+
+export class ReactExtensionMain {
+
+// ...
+
+  static dependencies = [ReactAspect, EnvsAspect];
+
+  static runtime = MainRuntime;
+
+  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+    const newReactEnv = react.compose([
+      react.overrideTsConfig(tsconfig)
+    ]);
+
+    envs.registerEnv(newReactEnv);
+
+    return new ReactExtensionMain(react, envs);
+
 }
+  ReactExtensionMain.addRuntime(ReactExtensionMain);
```

Comment:
should be `ReactWithProvidersAspect.addRuntime`

Reply:
Done.